Rename ContractProvider and ContractDirectory

maraoz opened the pull request:
Rename ContractProvider to ImplementationProvider and ContractDirectory to ImplementationDirectory

Works towards https://github.com/zeppelinos/zos-lib/issues/68

facuspagnuolo commented on contracts/application/management/UnversionedAppManager.sol:
```diff
@@ -28,15 +28,15 @@ contract UnversionedAppManager is BaseAppManager {
    * @dev Gets the contract provider used by the manager
    * @return Contract provider used by the manager
```
implementation provider

facuspagnuolo commented on contracts/application/management/UnversionedAppManager.sol:
```diff
@@ -10,14 +10,14 @@ import "../../upgradeability/UpgradeabilityProxyFactory.sol";
 
   /**
    * @dev Constructor function
    * @param _provider Contract provider
```
implementation provider

facuspagnuolo commented on contracts/application/management/AppDirectory.sol:
```diff
@@ -1,37 +1,39 @@
 
 /**
  * @title AppDirectory
  * @dev Contract directory with a standard library as a default provider
```
implementation directory

maraoz commented:
Updated! 

@spalladino I don't think merging this PR requires the other PRs first, no? Once we merge this, we can release a new version of `zos-lib`, and after that I'll work on updating `zos-cli` to this version with the new names.

@facuspagnuolo good catch! I did a project search for "contract provider" and "contract directory" and fixed all occurrences.

spalladino commented:
@maraoz absolutely, it can be merged without updating the dependency in the other repos. It was just a heads-up to ensure that we do reflect that rename in zos-cli.
